lighthouse-core/test/runner-test.js

```diff
@@ -295,4 +295,56 @@ describe('Runner', () => {
       assert.strictEqual(AuditClass.meta.name, auditExpectedName);
     });
   });
+
+  it('returns artifacts when audits are given', () => {
+    const url = 'https://example.com';
+    const config = new Config({
+      audits: [
+        'is-on-https'
+      ],
+
+      artifacts: {
+        HTTPS: {
+          value: true
+        }
+      }
+    });
+
+    return Runner.run({}, {url, config}).then(results => {
+      assert.ok(results.artifacts !== undefined);
+    });
+  });
+
+  it('returns artifacts when auditResult is given', () => {
+    const url = 'https://example.com';
+    const config = new Config({
+      auditResults: [{
+        name: 'is-on-https',
+        rawValue: true,
+        score: true,
+        displayValue: ''
+      }],
+
+      aggregations: [{
+        name: 'Aggregation',
+        description: '',
+        scored: true,
+        categorizable: true,
+        items: [{
+          name: 'name',
+          description: 'description',
+          audits: {
+            'is-on-https': {
+              expectedValue: true,
+              weight: 1
+            }
+          }
+        }]
+      }]
+    });
+
+    return Runner.run(null, {url, config, driverMock}).then(results => {
+      assert.ok(results.artifacts !== undefined);
+    });
+  });
```

Comment:
should probably have one more with `passes` and `audits` in the config file. `driverMock` will provide a trace, so could do a pass that records a trace and then an audit that purely requires a trace (maybe `user-timings`?). Something like
```js
const config = new Config({
  passes: [{
    recordTrace: true,
    gatherers: []
  }],
  audits: [
    'user-timings'
  ]
});
```
and then the result should have a `results.artifacts.traces` array with length > 0

That's kind of venturing into integration testing (relying on `gather-runner` internally, for one), but good to catch this case because most lighthouse runs are with configs of the passes/audits format. What do you think?